### PR TITLE
Os 56 Support Mime type (JSON/LD) in request & response.

### PR DESCRIPTION
**Pull request by pritha-tarento**
Summary Changes:
1.  Refactoring - frame Entity as a separate service 
2. Support Mime type in request for JSON/LD - request transformation impl as part of interceptor module. 
3. For request - required to add a content type as application/json for JSON Request and application/ld+json for JSON-LD request.
4. Support Mime type in request for JSON/LD - response transformation impl as a part of registry module.
5. For response - required to add Accept as application/json or application/ld+json for JSON or JSON-LD response.
6. Support of one mime type at a time while request/response present. 

Testing done using postman:
1. Adding an entity as JSON request & JSON-LD request 
2. Read an entity as JSON & JSON-LD response.

### CODE REVIEW

File: java/middleware-commons/src/main/java/io/opensaber/registry/middleware/transform/commoms/JsonUtils.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package io.opensaber.registry.middleware.transform.commoms;
```
Comment (indrajra) [inline java/middleware-commons/src/main/java/io/opensaber/registry/middleware/transform/commoms/JsonUtils.java]: Think this is not getting used at all. Also, for such one liners from standard library, please avoid creating wrappers.
Check the spelling of this parent directory - commons (n is replaced with m).

File: java/registry-interceptor/src/main/java/io/opensaber/registry/interceptor/request/transform/IRequestTransformer.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package io.opensaber.registry.interceptor.request.transform;
+
+import java.io.IOException;
+
+import io.opensaber.registry.middleware.transform.commoms.Data;
+import io.opensaber.registry.middleware.transform.commoms.TransformationException;
+
+
+public interface IRequestTransformer<T> {
```
Comment (indrajra) [inline java/registry-interceptor/src/main/java/io/opensaber/registry/interceptor/request/transform/IRequestTransformer.java]: Can we call this just as ITransformer?

File: java/registry-interceptor/src/main/java/io/opensaber/registry/interceptor/request/transform/RequestTransformFactory.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+import io.opensaber.registry.middleware.transform.commoms.ErrorCode;
+import io.opensaber.registry.middleware.transform.commoms.TransformationException;
+
+
+public class RequestTransformFactory {
+
+	private static final String MEDIATYPE_APPLICATION_JSONLD = "application/ld+json";
+	private static final String MEDIATYPE_APPLICATION_JSON = "application/json";
+	private static final String EXCEPTION_MESSAGE = "Media type not suppoted";
```
Comment (indrajra) [inline java/registry-interceptor/src/main/java/io/opensaber/registry/interceptor/request/transform/RequestTransformFactory.java]: Do spell check - supported.

File: java/registry-interceptor/src/main/java/io/opensaber/registry/interceptor/request/transform/RequestTransformFactory.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+	@Autowired
+	private JsonToLdRequestTransformer jsonToLdRequestTransformer;
+
+	@Autowired
+	private JsonldToLdRequestTransformer jsonldToLdRequestTransformer;
+
+	public IRequestTransformer<Object> getInstance(String type) throws TransformationException {
+		IRequestTransformer<Object> responseTransformer = null;
```
Comment (indrajra) [inline java/registry-interceptor/src/main/java/io/opensaber/registry/interceptor/request/transform/RequestTransformFactory.java]: Re-work on var, class names - request and response seems confusing to interpret.

File: java/registry/src/main/java/io/opensaber/registry/controller/RegistryController.java [hunk 5]
```diff
@@ -303,4 +321,14 @@
 		}
 		return new ResponseEntity<>(response, HttpStatus.OK);
 	}
+	
+	/*
+	 * To set the keys(like "@id" or @"@type" to be trim of a json
+	 */
+	private List<String> getKeysToTrim(){
```
Comment (indrajra) [inline java/registry/src/main/java/io/opensaber/registry/controller/RegistryController.java]: Though trim also means deleting, I think it makes more sense to use prune or delete. Just a suggestion.

File: java/registry/src/main/java/io/opensaber/registry/frame/FrameEntityImpl.java [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+import io.opensaber.registry.frame.FrameEntity;
+import io.opensaber.registry.middleware.util.Constants;
+import io.opensaber.registry.middleware.util.RDFUtil;
+
+@Component
+public class FrameEntityImpl implements FrameEntity {
+
+	private static Logger logger = LoggerFactory.getLogger(FrameEntityImpl.class);
+
+	@Value("${frame.file}")
```
Comment (indrajra) [inline java/registry/src/main/java/io/opensaber/registry/frame/FrameEntityImpl.java]: At some point down in the future, we might have more than one frame file. I see you've read frame.json in the transformation as well. Can you channelize all reads of this file from one class, please?

File: java/registry/src/main/java/io/opensaber/registry/transformation/JsonToLdTransformer.java [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+import java.text.ParseException;
+import java.util.*;
+
+@Component
+public class JsonToLdTransformer implements IResponseTransformer<Object> {
+
+	private static Logger logger = LoggerFactory.getLogger(JsonToLdTransformer.class);
+	private List<String> keysToTrim = new ArrayList<>();
+
+	public Data<Object> transform(Data<Object> data, List<String> keysToTrim) throws TransformationException {
```
Comment (indrajra) [inline java/registry/src/main/java/io/opensaber/registry/transformation/JsonToLdTransformer.java]: If you create another method to set the keysToTrim, you have a chance to combine IRequestTransformer and IResponseTransformer and simplify. No need to distinguish the direction in which the data is flowing - transformers shouldn't care about the direction, isn't it?

File: java/registry/src/main/java/io/opensaber/registry/transformation/JsonToLdTransformer.java [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+	private JsonNode addRootTypeNode(JsonNode graphNode){
+        String rootNodeType = graphNode.path(JsonldConstants.TYPE).asText();
+        ObjectNode rootNode = JsonNodeFactory.instance.objectNode();
+        rootNode.set(rootNodeType, graphNode);
+        return rootNode;
+		
+	}
+
+	private ObjectNode trimedKeyOfNodes(JsonNode node) {
+		ObjectNode result = JsonNodeFactory.instance.objectNode();
```
Comment (indrajra) [inline java/registry/src/main/java/io/opensaber/registry/transformation/JsonToLdTransformer.java]: I think we could use remove or removeAll API here - https://www.baeldung.com/jackson-json-node-tree-model

File: java/registry-interceptor/src/main/java/io/opensaber/registry/interceptor/request/transform/IRequestTransformer.java (hunk 2 above)
Comment (pritha-tarento) [reply java/registry-interceptor/src/main/java/io/opensaber/registry/interceptor/request/transform/IRequestTransformer.java]: This is name as specific to request in interceptor module.

File: java/middleware-commons/src/main/java/io/opensaber/registry/middleware/transform/commoms/JsonUtils.java (hunk 1 above)
Comment (pritha-tarento) [reply java/middleware-commons/src/main/java/io/opensaber/registry/middleware/transform/commoms/JsonUtils.java]: Right the file is not used I will remove it, Also name commons correctly

File: java/registry/src/main/java/io/opensaber/registry/transformation/JsonToLdTransformer.java (hunk 8 above)
Comment (pritha-tarento) [reply java/registry/src/main/java/io/opensaber/registry/transformation/JsonToLdTransformer.java]: Yes, have check this multiple time. Remove- remove by field name does not work for the nested objected. So it does not give me any help, hence had to iterate.

File: java/registry/src/main/java/io/opensaber/registry/transformation/JsonToLdTransformer.java (hunk 7 above)
Comment (pritha-tarento) [reply java/registry/src/main/java/io/opensaber/registry/transformation/JsonToLdTransformer.java]: Done

File: java/registry/src/main/java/io/opensaber/registry/frame/FrameEntityImpl.java (hunk 6 above)
Comment (pritha-tarento) [reply java/registry/src/main/java/io/opensaber/registry/frame/FrameEntityImpl.java]: Done

File: java/registry-interceptor/src/main/java/io/opensaber/registry/interceptor/request/transform/RequestTransformFactory.java (hunk 4 above)
Comment (pritha-tarento) [reply java/registry-interceptor/src/main/java/io/opensaber/registry/interceptor/request/transform/RequestTransformFactory.java]: Done

File: java/registry-interceptor/src/main/java/io/opensaber/registry/interceptor/request/transform/IRequestTransformer.java (hunk 2 above)
Comment (pritha-tarento) [reply java/registry-interceptor/src/main/java/io/opensaber/registry/interceptor/request/transform/IRequestTransformer.java]: Done

File: java/middleware-commons/src/main/java/io/opensaber/registry/middleware/transform/commoms/JsonUtils.java (hunk 1 above)
Comment (pritha-tarento) [reply java/middleware-commons/src/main/java/io/opensaber/registry/middleware/transform/commoms/JsonUtils.java]: Done

File: java/registry-interceptor/src/main/java/io/opensaber/registry/interceptor/request/transform/RequestTransformFactory.java (hunk 3 above)
Comment (pritha-tarento) [reply java/registry-interceptor/src/main/java/io/opensaber/registry/interceptor/request/transform/RequestTransformFactory.java]: Done

**Pull request merged**